zio-http/jvm/src/main/scala/zio/http/netty/server/ServerInboundHandler.scala

```diff
@@ -58,7 +58,15 @@ private[zio] final case class ServerInboundHandler(
   def refreshApp(): Unit = {
     val pair = appRef.get()
 
-    this.handler = pair._1.toHandler
+    this.handle = pair._1 match {
+      case Left(unscopedHandler) =>
+        val handler = unscopedHandler.toHandler
+        handler.apply
+      case Right(scopedHandler)  =>
+        val handler = scopedHandler.toHandler
+        (req: Request) => ZIO.scoped(handler(req))
```

Comment:
The scope lifetime must essentially span across https://github.com/jgranstrom/zio-http/blob/feat-scoped-requests/zio-http/jvm/src/main/scala/zio/http/netty/server/ServerInboundHandler.scala#L99 and https://github.com/jgranstrom/zio-http/blob/feat-scoped-requests/zio-http/jvm/src/main/scala/zio/http/netty/server/ServerInboundHandler.scala#L79, not sure on that one yet.